src/Compilers/CSharp/Test/Symbol/Compilation/ReferenceManagerTests.cs

```diff
@@ -2189,5 +2190,33 @@ public void ReusingCorLibManager()
             Assert.Same(assembly2.CorLibrary, assembly2);
             Assert.True(corlib1.ReferenceManagerEquals(corlib2));
         }
+
+        [WorkItem(43)]
+        [Fact]
+        public void AsymmetricUnification()
+        {
+            var vectors40 = CreateCompilationWithMscorlib(
+                @"[assembly: System.Reflection.AssemblyVersion(""4.0.0.0"")]",
+                options: TestOptions.ReleaseDll.WithCryptoPublicKey(TestResources.TestKeys.PublicKey_b03f5f7f11d50a3a),
+                assemblyName: "System.Numerics.Vectors");
+
+            Assert.Equal("System.Numerics.Vectors, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", vectors40.Assembly.Identity.GetDisplayName());
+
+            var vectors41 = CreateCompilationWithMscorlib(
+                @"[assembly: System.Reflection.AssemblyVersion(""4.1.0.0"")]",
+                options: TestOptions.ReleaseDll.WithCryptoPublicKey(TestResources.TestKeys.PublicKey_b03f5f7f11d50a3a),
+                assemblyName: "System.Numerics.Vectors");
+
+            Assert.Equal("System.Numerics.Vectors, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a", vectors41.Assembly.Identity.GetDisplayName());
+
+            var refVectors40 = vectors40.EmitToImageReference();
+            var refVectors41 = vectors41.EmitToImageReference();
+
+            var c1 = CreateCompilationWithMscorlib("", new[] { refVectors40, refVectors41 }, options: TestOptions.ReleaseDll.WithAssemblyIdentityComparer(DesktopAssemblyIdentityComparer.Default));
+            c1.VerifyDiagnostics();
```

Comment:
The first one didn't report any errors, the second one reported duplicate references with non-equivalent identity.